IdentityCore/src/parameters/MSIDRequestParameters.m

```diff
@@ -239,6 +249,34 @@ - (MSIDConfiguration *)msidConfiguration
     return _msidConfiguration;
 }
 
+- (void)updateAppRequestMetadata:(NSString *)homeAccountId
+{
+    MSIDAccountIdentifier *accountIdentifier = self.accountIdentifier;
+    
+    if (![NSString msidIsStringNilOrBlank:homeAccountId])
+    {
+        accountIdentifier = [[MSIDAccountIdentifier alloc] initWithDisplayableId:nil homeAccountId:homeAccountId];
+    }
+    
+    NSMutableDictionary *appRequestMetadata = [self.appRequestMetadata mutableCopy];
+    [appRequestMetadata removeObjectForKey:MSID_CCS_HINT_KEY];
```

Comment:
Should we ensure that either uid/utid or upn is present before cleaning up this key?

Reply:
sounds good to me.